### PR TITLE
move pipe setup to 'connect' event.

### LINKED ISSUE #9: Requests after reconnect don't complete
STR:
1- Create client pointing to server
2 - Make requests to bloomd  -- success
3- Stop and restart bloomd
4- Wait enough time so that client reconnects

Expected:
Requests to bloomd should succeed after reconnect

Actual:
Callback is never called

### PR DESCRIPTION
**Pull request by vinibaggio**
Hello @majelbstoat, 

Please review the following commits I made in branch 'vinny-fix-streams-on-reconnect'.

395d1b9b80f92a775eda8fd6c34605feafa7f5f9 (2015-10-28 17:02:05 -0700)
move pipe setup to 'connect' event.
[fixes #9]

R=@majelbstoat

### CODE REVIEW

**Comment by vinibaggio** [comment]
This is really hard to repro in a test environment, sorry for the lack of tests.

File: index.js [hunk 1]
```diff
@@ -459,6 +452,14 @@ BloomClient.prototype._onConnect = function () {
 
   this.unavailable = false
 
+  this.responseParser = new ResponseParser(this)
```
Comment (majelbstoat) [inline index.js]: nit: do we need a new ResponseParser for each connection? i.e. could we still do this.responseParser initialisation in the constructor?

**Comment by majelbstoat** [comment]
One question, otherwise LG

**Comment by majelbstoat** [comment]
Oh, and remember to bump package json version

**Comment by vinibaggio** [comment]
@majelbstoat from my understanding of nodejs streams, it seems we have to create a new stream:
```
By default end() is called on the destination when the source stream emits end, so that destination is no longer writable. Pass { end:
false } as options to keep the destination stream open.
```
It seems that this would imply that, once `end()` is called, the Writable is no longer writable.

**Comment by vinibaggio** [comment]
to better answer this question I'll just do a quick local test!

**Comment by vinibaggio** [comment]
This is what I get:
```
> Error: write after end
    at writeAfterEnd (_stream_writable.js:159:12)
    at ResponseParser.Writable.write (_stream_writable.js:204:5)
    at Socket.ondata (_stream_readable.js:528:20)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Socket.Readable.push (_stream_readable.js:110:10)
    at TCP.onread (net.js:523:20)
```

merging as it is!

**Pull request merged**